core/src/services/memcached/backend.rs

```diff
@@ -265,14 +325,85 @@ impl bb8::ManageConnection for MemcacheConnectionManager {
 
     /// TODO: Implement unix stream support.
     async fn connect(&self) -> Result<Self::Connection, Self::Error> {
-        let conn = TcpStream::connect(&self.address)
-            .await
-            .map_err(new_std_io_error)?;
-        let mut conn = binary::Connection::new(conn);
+        let conn = if self.enable_tls {
+            let mut root_cert_store = rustls::RootCertStore::empty();
+
+            let native_certs = rustls_native_certs::load_native_certs();
+            if native_certs.errors.is_empty() {
+                for cert in rustls_native_certs::load_native_certs().expect("unreachable!") {
+                    root_cert_store.add(cert).map_err(|err| {
+                        Error::new(ErrorKind::Unexpected, "tls connect failed").set_source(err)
+                    })?;
+                }
+            }
 
-        if let (Some(username), Some(password)) = (self.username.as_ref(), self.password.as_ref()) {
-            conn.auth(username, password).await?;
-        }
+            let config = if let (Some(cert_path), Some(key_path)) = (&self.tls_cert, &self.tls_key)
+            {
+                let cert_chain = CertificateDer::pem_file_iter(cert_path)
+                    .map_err(|err| {
+                        Error::new(ErrorKind::Unexpected, "tls connect failed").set_source(err)
+                    })?
+                    .filter_map(Result::ok)
+                    .collect();
+                let key_der = PrivateKeyDer::from_pem_file(key_path).map_err(|err| {
+                    Error::new(ErrorKind::Unexpected, "tls connect failed").set_source(err)
+                })?;
+
+                rustls::ClientConfig::builder()
```

Comment:
It's better for us to build this config once and use it for every connection. Loading and parsing certs is a heavy work.